### PR TITLE
Utility methods

### PR DESCRIPTION
**Pull request by hackaugusto**
added `__repr__` , `__eq__` and `__ne__` to most `State`, `StateChange` and `Event` classes

### CODE REVIEW

File: raiden/transfer/state_change.py [hunk 1]
```diff
@@ -7,33 +8,32 @@
...
 
 class Block(StateChange):
     """ Transition used when a new block is mined.
-
     Args:
         block_number: The current block_number.
     """
 
-    def __init__(self, block_number):
+    def __init__(self, block_number: int):
```
Comment (palango) [inline raiden/transfer/state_change.py]: This should be `typing.block_number`

**Comment by palango** [comment]
Looks good.

Just a side node: all these implementations feel like so much boilerplate and the perfect use case for dataclasses. However, there doesn't seem to be a implementation that will be compatible with Python 3.7. I only found https://github.com/ericvsmith/dataclasses

File: raiden/transfer/state_change.py [hunk 2]
```diff
@@ -252,12 +257,17 @@ def __ne__(self, other):
 class ContractReceiveChannelNewBalance(StateChange):
     """ A channel to which this node IS a participant had a deposit. """
 
-    def __init__(self, channel_identifier, participant_address: address, contract_balance: int):
-        if not isinstance(participant_address, address):
+    def __init__(
+            self,
+            channel_identifier,
+            participant_address: typing.address,
+            contract_balance: typing.block_number):
```
Comment (hackaugusto) [inline raiden/transfer/state_change.py]: humm, contract balance is not a block number, I will fix in an up coming pr

Comment (LefterisJP) [reply raiden/transfer/state_change.py]: argh copy pasta ... sorry

**Pull request merged**